librosa/util/utils.py

```diff
@@ -1205,6 +1205,48 @@ def localmin(x: np.ndarray, *, axis: int = 0) -> np.ndarray:
     return lmin
 
 
+
+@numba.guvectorize(
+    [
+        "void(float32[:], uint32, uint32, uint32, uint32, float32, uint32, bool_[:])",
+        "void(float64[:], uint32, uint32, uint32, uint32, float32, uint32, bool_[:])",
+        "void(int32[:], uint32, uint32, uint32, uint32, float32, uint32, bool_[:])",
+        "void(int64[:], uint32, uint32, uint32, uint32, float32, uint32, bool_[:])",
+    ],
+    "(n),(),(),(),(),(),()->(n)",
+    nopython=True, cache=True)
+def __peak_pick(x, pre_max, post_max, pre_avg, post_avg, delta, wait, peaks):
+    """Vectorized wrapper for the peak-picker"""
+    # Special case the first frame
+    peaks[0] = (x[0] >= np.max(x[:min(post_max, x.shape[0])]))
+    peaks[0] &= (x[0] >= np.mean(x[:min(post_avg, x.shape[0])]) + delta)
+
+    if peaks[0]:
+        n = wait + 1
+    else:
+        n = 1
+
+    while n < x.shape[0]:
+        maxn = np.max( x[max(0, n-pre_max):min(n+post_max, x.shape[0])])
+
+        # Are we the local max and sufficiently above average?
+        peaks[n] = (x[n] == maxn) 
+        
+        if not peaks[n]:
+            n += 1
+            continue
+
+        avgn = np.mean(x[max(0, n-pre_avg):min(n+post_avg, x.shape[0])])
```

Comment:
i'm going to trust you on this one!

Reply:
The computation here is pretty much line-for-line identical to the old (non-numba) implementation.

The only significant change to the algorithm is the variable increment to `n` (line 1247).  This implements the `wait` period between detections without having to do the intermediate computations on frames that we know will not trigger detection.  The output of the function is unchanged, but it's significantly faster.